rosetta/src/CMakeLists.txt

```diff
@@ -141,3 +141,35 @@ if(ROSETTA_PPM_SYCL)
     target_link_libraries(rosetta-sycl PUBLIC IntelSYCL::SYCL_CXX)
   endif()
 endif()
+
+if(ROSETTA_PPM_HIP)
+  add_library(rosetta-hip rosetta-hip.cpp ${_common_sources})
+  set_target_properties(rosetta-hip PROPERTIES FOLDER "Runtime")
+  target_compile_features(rosetta-hip PUBLIC cxx_std_17)
+  target_compile_features(rosetta-hip PUBLIC hip_std_17)
+
+  # Adding the feature cxx_std_17 does not forward to compiling the rosetta-hip
+  # library. Seems to be a CMake bug. CXX_STANDARD_REQUIRED also does not work
+  set_property(TARGET rosetta-hip PROPERTY CXX_STANDARD 17)
+
+  target_compile_definitions(rosetta-hip PUBLIC ROSETTA_PPM_HIP=1)
+  target_include_directories(rosetta-hip
+                             PUBLIC "${CMAKE_SOURCE_DIR}/rosetta/include")
+  if(HIP_FOUND)
+    # Currently, the rosetta-hip library has only host code and the executable
+    # that link to it consist only of .hip file which makes this split work.
+    # hip::device adds the --offload-arch flag that the C++ compiler (e.g. gcc)
+    # may not support. Ideally, we would compile the .hip files with the
+    # hip::device flags and the .cxx/.cpp files with the hip::host flags, no
+    # matter which target they belong to. TODO: File a bug with CMake/ROCm
```

Comment:
My comment is now outdated with `hip::device` gone. And `hip::host` can be applied to the `.hip` files (even though it is redundant), so there is not split between INTERFACE/PRIVATE anymore.